sw/airborne/modules/guidance/gvf/gvf.c

```diff
@@ -457,3 +476,259 @@ bool gvf_sin_wp_alpha(uint8_t wp, float alpha, float w, float off, float A)
   return true;
 }
 
+
+#elif defined(ROVER_FIRMWARE) // ROVER TRAJECTORIES
+
+// STRAIGHT LINE
+
+static void gvf_line(float a, float b, float heading)
+{
+  float e;
+  struct gvf_grad grad_line;
+  struct gvf_Hess Hess_line;
+
+  gvf_trajectory.type = 0;
+  gvf_trajectory.p[0] = a;
+  gvf_trajectory.p[1] = b;
+  gvf_trajectory.p[2] = heading;
+
+  gvf_line_info(&e, &grad_line, &Hess_line);
+  gvf_control.ke = gvf_line_par.ke;
+  gvf_control_2D(1e-2 * gvf_line_par.ke, gvf_line_par.kn, e, &grad_line, &Hess_line);
+
+  gvf_control.error = e;
+  
+  nav.mode = NAV_MODE_WAYPOINT;
+  
+  gvf_segment.seg = 0;
+}
+
+bool gvf_line_XY_heading(float a, float b, float heading)
+{
+  gvf_set_direction(1);
+  gvf_line(a, b, heading);
+  return true;
+}
+
+bool gvf_line_XY1_XY2(float x1, float y1, float x2, float y2)
+{
+  float zx = x2 - x1;
+  float zy = y2 - y1;
+
+  gvf_line_XY_heading(x1, y1, atan2f(zx, zy));
+  
+  nav.mode = NAV_MODE_ROUTE;
+  
+  gvf_segment.seg = 1;
+  gvf_segment.x1 = x1;
+  gvf_segment.y1 = y1;
+  gvf_segment.x2 = x2;
+  gvf_segment.y2 = y2;
+
+  // Send rover_base navigation data to draw segment (ROUTE)
+  nav_rover_base.goto_wp.from.x = x1;
+  nav_rover_base.goto_wp.from.y = y1;
+  nav_rover_base.goto_wp.to.x = x2;
+  nav_rover_base.goto_wp.to.y = y2;
+
+  return true;
+}
+
+bool gvf_line_wp1_wp2(uint8_t wp1, uint8_t wp2)
+{
+  float x1 = waypoints[wp1].enu_f.x;
```

Comment:
I recommend to use the `waypoint_get_x|y` function instead of accessing the element directly